File: csharp-reference/Characters/Frieza.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Frieza(string? name = null) : BaseFighter(name)
{
    public override string Id => "frieza";
    public override string DisplayName => "Frieza";
    public override string Greeting => "Try to entertain me.";
    public override string SignatureAttack => "Death Beam";
    public override string AttackStyle => "orb";
    public override double Scale => 0.93;
    public override FighterStats Stats => new(12000, 91, 82, 0.23, 22);
    public override FighterPalette Palette => new(
        "#f2f0f5", "#7136aa", "#f2f0f5", "#743ab2", "#bc8bf1", "#d678ff");
}

File: src/game/characters/Frieza.ts
import { BaseFighter } from "../BaseFighter";

export class Frieza extends BaseFighter {
  readonly id = "frieza" as const;
  readonly name = "Frieza";
  readonly greeting = "Try to entertain me.";
  readonly signatureAttack = "Death Beam";
  readonly attackStyle = "orb" as const;
  readonly power = 12000;
  readonly speed = 91;
  readonly defense = 82;
  readonly criticalChance = 0.23;
  readonly baseDamage = 22;
  readonly nativeFacing = "right" as const;
  readonly scale = 0.93;
  readonly palette = {
    skin: "#f2f0f5",
    hair: "#7136aa",
    primary: "#f2f0f5",
    secondary: "#743ab2",
    accent: "#bc8bf1",
    aura: "#d678ff"
  };
}
